File: ios/Swapl/Core/Repositories/SupportContactsRepository.swift
import Foundation

// GET /api/config/support-contacts — the 24/7 phone line and help-centre URL
// surfaced from the "Report a problem" flow. These used to be hardcoded in the
// app; now they come from one server endpoint so ops can change them without a
// release. Public, no auth.
//
// Best-effort: callers start from the launch defaults and overlay the server
// values once they load, so the UI never blocks on this lookup.
final class SupportContactsRepository: @unchecked Sendable {
    static let shared = SupportContactsRepository()

    func fetch() async throws -> SupportContacts {
        try await APIClient.shared.send("GET", "/api/config/support-contacts")
    }
}

struct SupportContacts: Decodable, Sendable, Equatable {
    let phone: String
    let helpUrl: String

    static let fallback = SupportContacts(
        phone: "+44 800 000 swap",
        helpUrl: "https://swapl.fun/help"
    )
}

File: android/swapl/app/src/main/java/app/swapl/core/repository/SupportContactsRepository.kt
package app.swapl.core.repository

import app.swapl.core.model.SupportContacts
import app.swapl.core.network.ApiClient
import io.ktor.client.call.body
import io.ktor.client.request.get
import javax.inject.Inject
import javax.inject.Singleton

// GET /api/config/support-contacts — public support config (24/7 phone line +
// help-centre URL) sourced from env on the server, with launch defaults. The
// dispute flow reads these so the "Report a problem" surfaces stay in sync with
// web and iOS without an app release.
@Singleton
class SupportContactsRepository @Inject constructor(private val api: ApiClient) {
    suspend fun fetch(): SupportContacts =
        api.client.get("${api.baseUrl}/api/config/support-contacts").body()
}
